File: backend/ai-service/app/prompts/goal_plan_generation.py
from app.prompts.base import BASE_SYSTEM_PROMPT


GOAL_PLAN_GENERATION_PROMPT = """
Create a realistic goal plan for a student.
Milestones must fit inside the requested timeline.
Next actions should be realistic, specific, and immediately doable.
Return JSON with summary, milestones, and exactly 3 to 5 nextActions.
""".strip()


def build_goal_plan_generation_instructions() -> str:
    return f"{BASE_SYSTEM_PROMPT}\n\n{GOAL_PLAN_GENERATION_PROMPT}"

File: backend/functions/src/ai/prompts/goal_plan_generation.ts
import { BASE_SYSTEM_PROMPT } from "./base.js";

export const GOAL_PLAN_GENERATION_SYSTEM_PROMPT = [
  BASE_SYSTEM_PROMPT,
  "Feature: goal_plan_generation.",
  "Purpose: turn one student goal into a reviewable plan with milestones and next actions.",
  "Behavior: create realistic milestones inside the requested timeline using the provided start date, timeline length, and timezone.",
  "Allowed scope: planning the goal, clarifying deliverables, sequencing work, and suggesting concrete next actions.",
  "Do not create impossible schedules, assume unavailable time, or claim the plan was saved to the user's planner.",
  "Output format: return summary, milestones, and nextActions only.",
  "Milestone dueDate values must be ISO8601 strings and must fit inside the requested timeline.",
  "Return exactly 3 to 5 practical nextActions.",
  "Each next action should be doable, specific, and sized for a student to start soon."
].join(" ");
